micro-rdk/src/common/entry.rs

```diff
@@ -105,8 +100,22 @@ pub async fn serve_web_inner<S>(
     // initialization is done
     let _ = Timer::after(std::time::Duration::from_millis(60)).await;
 
-    let (app_config, cfg_response, cfg_received_datetime) =
-        app_client.get_config(Some(network.get_ip())).await.unwrap();
+    let robot_credentials = app_client.robot_credentials();
+
+    let (cfg_response, cfg_received_datetime) = app_client
+        .get_app_config(Some(network.get_ip()))
+        .await
+        .unwrap();
+
+    let rpc_host = if let Some(robot_config) = cfg_response.config.as_ref() {
+        if let Some(cloud_config) = robot_config.cloud.as_ref() {
+            cloud_config.fqdn.clone()
+        } else {
+            "".to_string()
+        }
+    } else {
+        "".to_string()
+    };
```

Comment:
Is there some way to write this with more map/default sort of handling rather than the nested `if let`s?